inc/cobs.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <cstdint>
+#include <cstddef>
+
+/*
+Consistent Overhead Byte Stuffing (COBS) removes a specific value from a
+list of values. The removed value can then, for example, be used as a
+frame marker.
+
+Refer to: https://en.wikipedia.org/wiki/Consistent_Overhead_Byte_Stuffing
+
+COBS considerations:
+
+1. Configurable marker byte value. The current implementation assumes
```

Comment:
I think

> It just works (tm)

`[0, 1, 2, 3, 4, 5, 3, 6, 7, 8]` would be encoded as `[(4), 0, 1, 2, {3}, 4, 5, {4}, 6, 7, 8, 3]`. (n) means an offset of n where a 3 *is not* inserted. {n} means an offset of n where a 3 *is* inserted.

700 non 3 bytes will be encoded as `[(255), 254 non 3, (255), 254 non 3, (193), 192 non 3, 3]`. 

The alternative is to simply XOR the input with 0x03 before encoding.